Packet++/header/IPv4Layer.h

```diff
@@ -228,18 +228,18 @@ namespace pcpp
 		 * (meaning zeroed addresses - 0.0.0.0) will not be added to the returned list. If some error occurs during the parsing or the value is invalid an empty vector is returned
 		 * @return A vector of IPv4 addresses parsed from the IPv4 option value
 		 */
-		std::vector<IPv4Address> getValueAsIpList()
+		std::vector<IPv4Address> getValueAsIpList() const
 		{
 			std::vector<IPv4Address> res;
 
 			if (m_Data == NULL)
 				return res;
 
-			size_t dataSize =  getDataSize();
+			size_t dataSize = getDataSize();
 			if (dataSize < 2)
 				return res;
 
-			uint8_t valueOffset = (uint8_t)(1);
+			uint8_t valueOffset = 1;
```

Comment:
I think removing the casting might cause a compilation warning on some platfroms

Reply:
It's just a positive constant. There is no any reason for warning. But I can roll back the changes.

I have looked into the build logs and not found any warnings that concern the changes made in this PR.

Reply:
I think I added that casting for a reason, probably a compilation warning. But it was a long time ago so I don't quite remember on which platform it was...

Reply:
Ok. I will return back the old code.

Reply:
Done